packages/libs/client/src/signing/eckoWallet/eckoTypes.ts

```diff
@@ -0,0 +1,73 @@
+import { ICommand } from '@kadena/types';
+
+import { IQuicksignResponseOutcomes } from '../../signing-api/v1/quicksign';
+import { ISignFunction, ISingleSignFunction } from '../ISignFunction';
+
+import { connect, isConnected, isInstalled } from './eckoCommon';
+
+export type EckoStatus = 'success' | 'fail';
+
+/**
+ * Interface that describes the common functions to be used with Ecko Wallet
+ * @public
+ */
+export interface ICommonEckoFunctions {
+  isInstalled: typeof isInstalled;
+  isConnected: typeof isConnected;
+  connect: typeof connect;
+}
+/**
+ * Interface to use when writing a signing function for Ecko Wallet that accepts a single transaction
+ * @public
+ */
+export interface IEckoSignSingleFunction
+  extends ISingleSignFunction,
+    ICommonEckoFunctions {}
```

Comment:
mixing a function with properties might lead to a confusing api for users since function itself has some methods like apply, call ,... 
what if we instead add the sign as a method like this
```ts
export interface IEckoSignFunction {
   isInstalled: typeof isInstalled;
   isConnected: typeof isConnected;
   connect: typeof connect;
   sign : ISingleFunction,
   quickSign : ISingleFunction,
}

Reply:
We've thought about that, but we wanted to make the usage of this function the same as the other functions to allow drop-in replacement.